windows/src/developer/TIKE/child/UfrmDebug.pas

```diff
@@ -630,15 +654,68 @@ procedure TfrmDebug.ExecuteEventRule(n: Integer);
 
 
 procedure TfrmDebug.ExecuteEventAction(n: Integer);
+  type
```

Comment:
All this relates to:
(1) Correct handling of text selection and deletion
(3) Deadkey realignment after text changes